src/org/rascalmpl/util/maven/Artifact.java

```diff
@@ -118,10 +119,12 @@ public IList getMessages() {
      * @return
      */
     public List<Artifact> resolveDependencies(Scope forScope, MavenParser parser) {
-        Set<WithoutVersion> alreadyIncluded = new HashSet<>();
+        Set<ResolveKey> alreadyResolved = new HashSet<>();
+        Map<ArtifactCoordinate.WithoutVersion, String> resolvedVersions = new HashMap<>();
+        Set<WithoutVersion> exclusions = new HashSet<>();
```

Comment:
2 lines below each other refer to `WithoutVersion` in 2 different ways? can we normalize this?